(DOCS-14141) New AI Costs page for CCM (#36431)
DataDog/documentation: content/en/cloud_cost_management/ai_costs.md

---
title: AI Costs
description: "Gain unified visibility into AI spend across providers, normalize cost data, and attribute usage to users and teams."
further_reading:
- link: "/cloud_cost_management/"
  tag: "Documentation"
  text: "Cloud Cost Management"
- link: "/cloud_cost_management/setup/saas_costs"
  tag: "Documentation"
  text: "SaaS Cost Integrations"
- link: "/cloud_cost_management/allocation/custom_allocation_rules"
  tag: "Documentation"
  text: "Custom Allocation Rules"
- link: "/cloud_cost_management/allocation/tag_pipelines"
  tag: "Documentation"
  text: "Tag Pipelines"
- link: "/cloud_cost_management/reporting"
  tag: "Documentation"
  text: "Reporting"
- link: "/cloud_cost_management/cost_changes/monitors"
  tag: "Documentation"
  text: "Cloud Cost Monitors"
- link: "/cloud_cost_management/planning/budgets"
  tag: "Documentation"
  text: "Budgets"
- link: "/cloud_cost_management/planning/forecasting"
  tag: "Documentation"
  text: "Forecasting"
---

## Overview

> AI cost visibility gives FinOps and engineering teams a unified destination for analyzing AI spend across providers, including Anthropic, AWS Bedrock, Google Gemini, OpenAI, and Vertex AI. View total AI spend alongside your existing cloud infrastructure costs, analyze it with normalized tags, track cost anomalies, and attribute usage to the specific users and API keys driving it.

Reviewer: Consider opening with the canonical product name "AI Costs" -- both AI agents and readers will land on this page and immediately know what the feature is.

> AI cost visibility gives FinOps and engineering teams a unified destination for analyzing AI spend across providers, including Anthropic, AWS Bedrock, Google Gemini, OpenAI, and Vertex AI. View total AI spend alongside your existing cloud infrastructure costs, analyze it with normalized tags, track cost anomalies, and attribute usage to the specific users and API keys driving it.

Reviewer: Also, Vale is flagging that "AWS Bedrock" should be "Amazon Bedrock"

## Prerequisites

To use AI Costs, you must have at least one of the following supported providers set up for [Cloud Cost Management][1]:

> | AI Provider | Setup method |

Reviewer: ```suggestion
| AI provider | Setup method |
```

| AWS Bedrock | [AWS integration][3] |
| Gemini      | [Google Cloud integration][4] |
| OpenAI     | [SaaS integration][5] |
| Vertex AI  | [Google Cloud integration][4] |

## AI cost summary

After connecting your AI providers, navigate to [**Cloud Cost** > **Summarize** > **AI**][6] to view the AI cost summary page.

{{< img src="cloud_cost/ai_costs/ccm-ai-costs-overview.png" alt="The AI cost summary dashboard, showing daily spend trends over a one-month period, a list of top spend drivers, and an anomaly graph." responsive="true" style="width:100%; background:none; border:none; box-shadow:none;">}}

The AI cost summary page provides:

- **Total AI Cost:** Aggregated AI cost and cost change over the selected time frame.
- **Daily AI Cost:** Daily cost trends across the selected providers over the selected time frame. Use the **Filter to** dropdown to define which providers appear in the graph.
- **Top Cost Drivers:** The models, projects, services, and users generating the most spend.
- **Active AI Cost Anomalies:** Cost [anomalies][7] surfaced proactively across all connected providers. Select an anomaly to open a side panel with more details and options for further action.

> - **AI Cost Dashboards:** Out-of-the-box dashboard templates for each supported provider, combining cost data with usage signals such as token consumption, model distribution, and user analytics.

Reviewer: ```suggestion
- **Total AI Cost**: Aggregated AI cost and cost change over the selected time frame.
- **Daily AI Cost**: Daily cost trends across the selected providers over the selected time frame. Use the **Filter to** dropdown to define which providers appear in the graph.
- **Top Cost Drivers**: The models, projects, services, and users generating the most spend.
- **Active AI Cost Anomalies**: Cost [anomalies][7] surfaced proactively across all connected providers. Select an anomaly to open a side panel with more details and options for further action.
- **AI Cost Dashboards**: Out-of-the-box dashboard templates for each supported provider, combining cost data with usage signals such as token consumption, model distribution, and user analytics.
```

## Normalized AI tags

> AI cost data from all supported providers is normalized to a consistent set of tags, so you can filter, group, compare, and plan AI spend consistently across dashboards, [monitors][8], [budgets][9], [forecasts][10], and other Datadog tools. Use the [Cloud Cost Explorer][11] to query and compare spend across providers without writing per-provider logic.

Reviewer: "consistent set of tags... consistently" repeats within the same sentence, and the sentence is ~40 words. Consider splitting:

```suggestion
AI cost data from all supported providers is normalized to a consistent set of tags. You can then filter, group, compare, and plan AI spend across dashboards, [monitors][8], [budgets][9], [forecasts][10], and other Datadog tools.
```

The following tags are available for all supported AI providers:

> | Tag Name | Tag Description |

Reviewer: ```suggestion
| Tag name | Tag description |
```

> | Tag Name | Tag Description |

Reviewer: Visual nit: the `Tag name` column is narrow enough that some tag names wrap onto two lines. Consider widening it — e.g., pad column 1 in the source so the markdown renderer allocates more space, or add a non-breaking spacer in the header.

|---|---|
| `providername` | The AI provider. |
| `model` | The AI model identifier (for example, `claude-opus-4-6`, `gpt-4.1`). |
| `model_name` | The human-readable model name (for example, `Claude Opus 4.6`). |
| `token_direction` | Whether tokens are being consumed (input) or generated (output) within a service or application. |
| `token_category` | The specific category of tokens consumed, such as input/output tokens or tokens related to caching and search operations (for example, `cached input`, `cache write`, `standard input`, `output`). |
| `project` | The project, workspace, or environment the AI costs belong to. |

## Attribute AI spend to users

> [Out-of-the-box (OOTB) allocation rules][12] use Datadog observability data to attribute AI costs to the users, API keys, and other sources that generated them. OOTB allocation rules require no configuration and are available for OpenAI and Anthropic.

Reviewer: Provider order: the prereq table lists Anthropic before OpenAI — flip "OpenAI and Anthropic" → "Anthropic and OpenAI" for consistency.

The following tags are available through OOTB allocation rules:

{{< tabs >}}
{{% tab "Anthropic" %}}

- `api_key_id`
- `api_key_name`
- `context_window`
- `model`
- `model_id`
- `org_id`
- `org_name`
- `service_tier`
- `user_email`
- `user_id`
- `user_name`
- `workspace_id`
- `workspace_name`

{{% /tab %}}
{{% tab "OpenAI" %}}

- `account_id`
- `account_name`
- `api_key_id`
- `batch`
- `endpoint`
- `model`
- `org_id`
- `project_id`
- `project_name`
- `user_email`
- `user_id`

{{% /tab %}}
{{< /tabs >}}

> {{< img src="cloud_cost/ai_costs/ccm-tag-pipeline-ai-costs.png" alt="The Tag Pipelines Rule Setup page, showing user_email values mapped to team values via an existing reference table, and additional tag mapping options." responsive="true" style="width:100%; background:none; border:none; box-shadow:none;">}}

Reviewer: Vale: `via` → `through`.

```suggestion
{{< img src="cloud_cost/ai_costs/ccm-tag-pipeline-ai-costs.png" alt="The Tag Pipelines Rule Setup page, showing user_email values mapped to team values through an existing reference table, and additional tag mapping options." responsive="true" style="width:100%; background:none; border:none; box-shadow:none;">}}
```

> After costs are attributed to users, API keys, and other sources, use [Tag Pipelines][13] to automatically map them to teams, services, or business units for streamlined aggregate reporting. See [Cost Reports][14] for more information.

Reviewer: "streamlined" is on the buzzword list — suggest dropping it.

```suggestion
After costs are attributed to users, API keys, and other sources, use [Tag Pipelines][13] to automatically map them to teams, services, or business units for aggregate reporting. See [Cost Reports][14] for more information.
```

> After costs are attributed to users, API keys, and other sources, use [Tag Pipelines][13] to automatically map them to teams, services, or business units for streamlined aggregate reporting. See [Cost Reports][14] for more information.

Reviewer: Flow in this section feels a little off — the two screenshots are sandwiching this one sentence. Both screenshots tie to it (setting up the tag pipeline, then seeing the resulting report), but the current order makes that connection hard to see.

Consider adding a transition after the tab list, then a short lead-in before each screenshot so the reader knows what they're about to look at. Something like:
- After the tabs: a sentence introducing the "next step" of mapping those tags to teams via Tag Pipelines.
- Before screenshot 1: a phrase pointing at setup (e.g., "Configure a Tag Pipelines rule to map `user_email` values to teams:").
- Before screenshot 2: a phrase pointing at the result (e.g., "Once mapped, attributed spend appears in provider-specific dashboards:").

> {{< img src="cloud_cost/ai_costs/ccm-anthropic-ai-cost-reporting.png" alt="A provider-specific dashboard with a stacked bar chart graph showing daily provider spend attributed by team and model name, and a summary list of the spend attributions." responsive="true" style="width:100%; background:none; border:none; box-shadow:none;">}}

Reviewer: "stacked bar chart graph" is redundant — use either "stacked bar chart" or "stacked bar graph."

```suggestion
{{< img src="cloud_cost/ai_costs/ccm-anthropic-ai-cost-reporting.png" alt="A provider-specific dashboard with a stacked bar chart showing daily provider spend attributed by team and model name, and a summary list of the spend attributions." responsive="true" style="width:100%; background:none; border:none; box-shadow:none;">}}
```

  tag: "Documentation"
  text: "Custom Allocation Rules"
- link: "/cloud_cost_management/allocation/tag_pipelines"
  tag: "Documentation"
  text: "Tag Pipelines"
- link: "/cloud_cost_management/reporting"
  tag: "Documentation"
  text: "Reporting"
- link: "/cloud_cost_management/cost_changes/monitors"
  tag: "Documentation"
  text: "Cloud Cost Monitors"
- link: "/cloud_cost_management/planning/budgets"
  tag: "Documentation"
  text: "Budgets"
- link: "/cloud_cost_management/planning/forecasting"
  tag: "Documentation"
  text: "Forecasting"
---

## Overview

AI Costs in Cloud Cost Management gives FinOps and engineering teams a unified destination for analyzing AI spend across providers, including Anthropic, Amazon Bedrock, Google Gemini, OpenAI, and Vertex AI. View total AI spend alongside your existing cloud infrastructure costs, analyze it with normalized tags, track cost anomalies, and attribute usage to the specific users and API keys driving it.

## Prerequisites

To use AI Costs, you must have at least one of the following supported providers set up for [Cloud Cost Management][1]:

| AI provider | Setup method |
|---|---|

> | Anthropic   | [SaaS integration][2] |

Reviewer: Nit: The prereq table's row order doesn't match the overview paragraph's order, and isn't strictly alphabetical either. Obvi not a dealbreaker, but my inner perfectionist had to flag :)

| Amazon Bedrock | [AWS integration][3] |

> | Gemini      | [Google Cloud integration][4] |

Reviewer: The overview paragraph says "Google Gemini" so matching here.

```suggestion
| Google Gemini      | [Google Cloud integration][4] |
```

{{< img src="cloud_cost/ai_costs/ccm-ai-costs-overview.png" alt="The AI cost summary dashboard, showing daily spend trends over a one-month period, a list of top spend drivers, and an anomaly graph." responsive="true" style="width:100%; background:none; border:none; box-shadow:none;">}}

The AI cost summary page provides:

- **Total AI Cost**: Aggregated AI cost and cost change over the selected time frame.
- **Daily AI Cost**: Daily cost trends across the selected providers over the selected time frame. Use the **Filter to** dropdown to define which providers appear in the graph.
- **Top Cost Drivers**: The models, projects, services, and users generating the most spend.
- **Active AI Cost Anomalies**: Cost [anomalies][7] surfaced proactively across all connected providers. Select an anomaly to open a side panel with more details and options for further action.
- **AI Cost Dashboards**: Out-of-the-box dashboard templates for each supported provider, combining cost data with usage signals such as token consumption, model distribution, and user analytics.

## Normalized AI tags

> AI cost data from all supported providers is normalized to a consistent set of tags. Use these tags to filter, group, compare, and plan AI spend across dashboards, [monitors][8], [budgets][9], [forecasts][10], and other Datadog tools. The [Cloud Cost Explorer][11] allows you to query and compare spend across providers without writing per-provider logic.

Reviewer: ```suggestion
AI cost data from all supported providers is normalized to a consistent set of tags. Use these tags to filter, group, compare, and plan AI spend across dashboards, [monitors][8], [budgets][9], [forecasts][10], and other Datadog tools. Use [Cloud Cost Explorer][11] to query and compare spend across providers without writing per-provider logic.
```

The following tags are available for all supported AI providers:

| Tag&nbsp;name&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; | Tag description |
|---|---|

> | `providername` | The AI provider. |

Reviewer: `providername` (no underscore) looks unusual next to its snake_case neighbors (`model_name`, `token_direction`, `token_category`). Just confirming the real tag key is `providername` and not `provider_name`?

| `model` | The AI model identifier (for example, `claude-opus-4-6`, `gpt-4.1`). |
| `model_name` | The human-readable model name (for example, `Claude Opus 4.6`). |
| `token_direction` | Whether tokens are being consumed (input) or generated (output) within a service or application. |

> | `token_category` | The specific category of tokens consumed, such as input/output tokens or tokens related to caching and search operations (for example, `cached input`, `cache write`, `standard input`, `output`). |

Reviewer: To make Vale happy.

```suggestion
| `token_category` | The specific category of tokens consumed, such as I/O tokens or tokens related to caching and search operations (for example, `cached input`, `cache write`, `standard input`, `output`). |
```

The following tags are available through OOTB allocation rules:

{{< tabs >}}
{{% tab "Anthropic" %}}

- `api_key_id`
- `api_key_name`
- `context_window`
- `model`
- `model_id`
- `org_id`
- `org_name`
- `service_tier`
- `user_email`
- `user_id`
- `user_name`
- `workspace_id`
- `workspace_name`

{{% /tab %}}
{{% tab "OpenAI" %}}

- `account_id`
- `account_name`
- `api_key_id`
- `batch`
- `endpoint`
- `model`
- `org_id`
- `project_id`
- `project_name`
- `user_email`
- `user_id`

{{% /tab %}}
{{< /tabs >}}

Configure [Tag Pipelines][13] to map OOTB tags (such as `user_email`) to teams, services, or business units for aggregate reporting:

{{< img src="cloud_cost/ai_costs/ccm-tag-pipeline-ai-costs.png" alt="The Tag Pipelines Rule Setup page, showing user_email values mapped to team values through an existing reference table, and additional tag mapping options." responsive="true" style="width:100%; background:none; border:none; box-shadow:none;">}}

> Once mapped, attributed spend appears in provider-specific dashboards and [Cost Reports][14]:

Reviewer: I introduced this one, my bad. Fixing to make Vale happy.

```suggestion
After mapping, attributed spend appears in provider-specific dashboards and [Cost Reports][14]:
```

## Normalized AI tags

AI cost data from all supported providers is normalized to a consistent set of tags, so you can filter, group, and compare AI spend consistently across dashboards, monitors, and other Datadog tools. Use the [Cloud Cost Explorer][6] to query and compare spend across providers without writing per-provider logic.

The following tags are available for all supported AI providers:

| Tag Name | Tag Description |
|---|---|
| `providername` | The AI provider. |
| `model` | The AI model identifier (for example, `claude-opus-4-6`, `gpt-4.1`). |
| `model_name` | The human-readable model name. |
| `token_direction` | Whether tokens are being consumed (input) or generated (output) within a service or application. |
| `token_category` | The specific category of tokens consumed, such as input/output tokens or tokens related to caching and search operations (for example, `cached input`, `cache write`, `standard input`, `output`). |
| `project` | The project, workspace, or environment the AI costs belong to. |
| `usage_type` | A combined charge description containing model, direction, cache, and other usage details. |

## Attribute AI spend to users

> Out-of-the-box (OOTB) allocation rules use Datadog observability data to attribute AI costs to the users and API keys that generated them. OOTB allocation rules require no configuration and are available for OpenAI.

Reviewer 2: available for OpenAI and Anthropic

---
title: AI Costs
description: "Gain unified visibility into AI spend across providers, normalize cost data, and attribute usage to users and teams."
further_reading:
- link: "/cloud_cost_management/"
  tag: "Documentation"
  text: "Cloud Cost Management"
- link: "/cloud_cost_management/setup/saas_costs"
  tag: "Documentation"
  text: "SaaS Cost Integrations"
- link: "/cloud_cost_management/allocation/custom_allocation_rules"
  tag: "Documentation"
  text: "Custom Allocation Rules"
- link: "/cloud_cost_management/allocation/tag_pipelines"
  tag: "Documentation"
  text: "Tag Pipelines"
---

## Overview

AI cost visibility gives FinOps and engineering teams a unified destination for analyzing AI spend across providers, including Anthropic, AWS Bedrock, Google Gemini, OpenAI, and Vertex AI. View total AI spend alongside your existing cloud infrastructure costs, analyze it with normalized tags, track cost anomalies, and attribute usage to the specific users and API keys driving it.

## Prerequisites

To use AI Cost Management, you must:

> - Be using [Cloud Cost Management][1] with a supported cloud provider ([AWS][8] or [Google Cloud][9]) or a connected [SaaS Cost Integration][2].

Reviewer 2: for the SaaS Cost Integration, we only support OpenAI and Anthropic for the dedicated AI Costs Summary page. should we combine these two bullet points, or have one for the clouds and another for the SaaS providers?

  - [OpenAI][2]
  - [Anthropic][3]

## AI cost summary

After connecting your AI providers, navigate to [**Cloud Cost** > **Summarize** > **AI**][5] to view the AI cost summary page.

The AI cost summary page provides:

- **Total AI Cost:** Aggregated AI cost and cost change over the selected time frame.
- **Daily AI Cost:** Daily cost trends across the selected providers over the selected time frame. Use the **Filter to** dropdown to define which providers appear in the graph.
- **Top Cost Drivers:** The models, projects, services, and users generating the most spend.
- **Active AI Cost Anomalies:** Cost [anomalies][4] surfaced proactively across all connected providers. Select an anomaly to open a side panel with more details and options for further action.
- **AI Cost Dashboards:** Out-of-the-box dashboard templates for each supported provider, combining cost data with usage signals such as token consumption, model distribution, and user analytics.

## Normalized AI tags

> AI cost data from all supported providers is normalized to a consistent set of tags, so you can filter, group, compare, and plan AI spend consistently across dashboards, monitors, budgets, and other Datadog tools. Use the [Cloud Cost Explorer][6] to query and compare spend across providers without writing per-provider logic.

Reviewer 2: - link to monitors documentation https://docs.datadoghq.com/cloud_cost_management/cost_changes/monitors?tab=costmetricbased
- link to budgets documentation https://docs.datadoghq.com/cloud_cost_management/planning/budgets?tab=basic
- after budgets, and "forecasts" and link to forecasting documentation https://docs.datadoghq.com/cloud_cost_management/planning/forecasting

The following tags are available for all supported AI providers:

| Tag Name | Tag Description |
|---|---|
| `providername` | The AI provider. |
| `model` | The AI model identifier (for example, `claude-opus-4-6`, `gpt-4.1`). |
| `model_name` | The human-readable model name (for example, `Claude Opus 4.6`). |
| `token_direction` | Whether tokens are being consumed (input) or generated (output) within a service or application. |
| `token_category` | The specific category of tokens consumed, such as input/output tokens or tokens related to caching and search operations (for example, `cached input`, `cache write`, `standard input`, `output`). |
| `project` | The project, workspace, or environment the AI costs belong to. |

> Additional normalized tags include:

Reviewer 2: i think these tags need to be moved to the "attribute" section below, because the OOTB rules for these tags are only available for OpenAI. we can add, "available for OpenAI, with more providers planned." ?

Reviewer 2: we will also be supporting Anthropic - some of the tags overlap with OpenAI and others are specific to Anthropic. let's decide if we need to show this in a table to distinguish between the two?
